src/controllers/activity.js

```diff
@@ -1,6 +1,72 @@
 const mongoose = require('mongoose');
 const Activity = require('../models/activity');
 
+const getAllActivities = (req, res) => {
+  Activity.find()
+    .then((activities) => {
+      if (activities.length === 0) {
+        return res.status(404).json({
+          message: 'There are no activities',
+          error: false,
+        });
+      }
+      return res.status(200).json({
+        message: 'Complete activities list',
+        data: activities,
+        error: false,
+      });
+    })
+    .catch((error) => res.status(500).json({ message: 'An error ocurred', error }));
+};
+
+const getActivitiesById = (req, res) => {
+  const { id } = req.params;
+  if (!mongoose.Types.ObjectId.isValid(id)) {
+    return res.status(400).json({
+      message: `Invalid activity id: ${id}`,
+      error: true,
+    });
+  }
+  return Activity.findById(id)
+    .then((activity) => {
+      if (activity == null) {
```

Comment:
Here you could change it to (!activity) like you did in line 83. This to avoid any problem if in some case activity = undefined.